src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -39,24 +39,29 @@ namespace System.IO.Enumeration
             // IMPORTANT: Attribute logic must match the logic in FileStatus
 
             bool isDirectory = false;
+            bool isSymlink = false;
             if (directoryEntry.InodeType == Interop.Sys.NodeType.DT_DIR)
             {
                 // We know it's a directory.
                 isDirectory = true;
             }
+            // Some operating systems don't have the inode type in the dirent structure,
+            // so we use DT_UNKNOWN as a sentinel value. As such, check if the dirent is a
+            // directory or a symlink.
             else if ((directoryEntry.InodeType == Interop.Sys.NodeType.DT_LNK
                 || directoryEntry.InodeType == Interop.Sys.NodeType.DT_UNKNOWN)
                 && Interop.Sys.Stat(entry.FullPath, out Interop.Sys.FileStatus targetStatus) >= 0)
             {
                 // Symlink or unknown: Stat to it to see if we can resolve it to a directory.
                 isDirectory = (targetStatus.Mode & Interop.Sys.FileTypes.S_IFMT) == Interop.Sys.FileTypes.S_IFDIR;
+                isSymlink = (targetStatus.Mode & Interop.Sys.FileTypes.S_IFMT) == Interop.Sys.FileTypes.S_IFLNK;
             }
 
             entry._status = default;
             FileStatus.Initialize(ref entry._status, isDirectory);
 
             FileAttributes attributes = default;
-            if (directoryEntry.InodeType == Interop.Sys.NodeType.DT_LNK)
+            if (isSymlink)
```

Comment:
good point; would adding an else if to the directory/stat if block for `DT_LNK` suffice?

Reply:
The behaviour I *do* see changing now though, is a workingsymlink to a directory, as before it would be, if I read it right, Reparse and Directory, whereas now it seemingly will just be just directory because the initial call to stat will resolve and succeed; actually, now that I think of it, the logic could be the old one, just repeated for directory and symlink?

Reply:
> The behaviour I do see changing now though, is a workingsymlink to a directory, as before it would be, if I read it right, Reparse and Directory, whereas now it seemingly will just be just directory because the initial call to stat will resolve and succeed

If that's changing from Reparse/Directory to just Directory, that's a problem that needs to be fixed.  I thought we had tests for that.

Reply:
No, I was correcting a change in this branch to bring it back to normal behaviour. The logic here should be the same as master when handling a link/directory, it just handles `DT_UNKNOWN` now.